riposte-core/src/main/java/com/nike/riposte/server/channelpipeline/HttpChannelInitializer.java

```diff
@@ -430,9 +451,9 @@ public void initChannel(SocketChannel ch) {
                       new OpenChannelLimitHandler(openChannelsGroup, maxOpenChannelsThreshold));
         }
 
-        // INBOUND - Add the RequestFilterHandler (if we have any filters to apply).
-        if (cachedRequestFilterHandler != null)
-            p.addLast(REQUEST_FILTER_HANDLER_NAME, cachedRequestFilterHandler);
+        // INBOUND - Add the RequestFilterHandler for before security (if we have any filters to apply).
+        if (beforeSecurityRequestFilterHandler != null)
+            p.addLast(REQUEST_FILTER_HANDLER_NAME, beforeSecurityRequestFilterHandler);
```

Comment:
We can't use the same handler name for both the before-security and after-security request filter handlers. According to the `ChannelPipeline` docs for `addLast(String, ChannelHandler)` you should get a `IllegalArgumentException` thrown if you try to add a handler with a name that's already in the pipeline.

`REQUEST_FILTER_HANDLER_NAME` should probably be split into two: `REQUEST_FILTER_BEFORE_SECURITY_HANDLER_NAME` and `REQUEST_FILTER_AFTER_SECURITY_HANDLER_NAME`.

And if you never received an `IllegalArgumentException` then that indicates a problem with the tests - we should have tests that validate the four cases: (1) no filters at all, (2) only before-security filters, (3) only after-security filters, and (4) both before-security and after-security filters.